Sources/OTelCore/OTel+Configuration+DiagnosticLogger.swift

```diff
@@ -0,0 +1,43 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2025 the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import class Foundation.ProcessInfo
+package import Logging
+
+extension OTel.Configuration {
+    package func makeDiagnosticLogger() -> Logger {
+        var logger = switch self.diagnosticLogger.backing {
+        case .console:
+            Logger(label: "swift-otel", factory: { label in StreamLogHandler.standardError(label: label) })
+        case .custom(let logger):
+            // TODO: would a better configuration API accept a custom factory, which would allow per-logger labels?
```

Comment:
Worth discussing I suppose, but I think the source location included in the logs already provides enough information to understand which part of OTel produced a certain log message.